src/test/java/io/jenkins/plugins/oidc_provider/IdTokenFileCredentialsTest.java

```diff
@@ -103,6 +119,31 @@ public class IdTokenFileCredentialsTest {
             parseClaimsJws(idToken).
             getBody();
         System.out.println(claims);
+        assertEquals("my_git_revision", claims.get("git_revision", String.class));
+        assertEquals("my_git_branch", claims.get("git_branch", String.class));
+    }
+
+    @Extension
+    public static class GitTokenMacro extends TokenMacro {
```

Comment:
Remove `token-macro` dep and you could use `EnvironmentContributor` for testing.

Reply:
For what said above, I'd be keen stick with token macro, given it seems the default way plugins are intended to exchange data (more [here](https://plugins.jenkins.io/token-macro/#plugin-content-motivation))

Reply:
> the default way plugins are intended to exchange data

In the world of traditional job types like Freestyle. Not normally used in Pipeline.